File: app/src/main/java/com/example/easyhttp/MainActivity.java
package com.example.easyhttp;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.TextView;

import com.example.easyhttp.http.HttpManager;
import com.example.easyhttp.http.callback.OnResultCallBack;
import com.example.easyhttp.http.subscriber.HttpSubscriber;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = MainActivity.class.getSimpleName();
    private HttpSubscriber mHttpObserver;
    private TextView resultTv;
    private String result;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        resultTv = (TextView) findViewById(R.id.result);
        findViewById(R.id.btn).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                result="";
                resultTv.setText("");
                HttpManager.getInstance().getDatasWithCache(mHttpObserver, 1, 10, "json", false);
            }
        });

        mHttpObserver = new HttpSubscriber(new OnResultCallBack<TestBean>() {
            @Override
            public void onSuccess(TestBean tb) {
                for (TestBean.ListBean bean : tb.getList()) {
                    result += bean.toString();
                }
                resultTv.setText(result);
            }

            @Override
            public void onError(int code, String errorMsg) {
                resultTv.setText("onError: code:" + code + "  errorMsg:" + errorMsg);
            }
        });
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mHttpObserver.unSubscribe();
    }
}

File: app/src/main/java/com/example/easyhttp/MainActivity.kt
package com.example.easyhttp

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.example.easyhttp.http.cache.RxCache
import com.example.easyhttp.http.cache.diskconverter.GsonDiskConverter
import com.example.easyhttp.http.callback.OnResultCallBack
import com.example.easyhttp.http.subscriber.HttpSubscriber
import com.knightdavion.kotlin.ibiliplayer.data.remote.HttpManager
import kotlinx.android.synthetic.main.activity_main.*
import java.io.File

class MainActivity : AppCompatActivity() {
    private var resultStr: String? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        RxCache.initializeDefault(RxCache.Builder()
                .appVersion(1)
                .diskDir(File(cacheDir.path + File.separator + "data-cache"))
                .diskConverter(GsonDiskConverter())
                .diskMax((20 * 1024 * 1024).toLong())
                .memoryMax(2 * 1024 * 1024)
                .setDebug(true)
                .build())

        btn.setOnClickListener {
            resultStr = ""
            HttpManager.getDatasCached(HttpSubscriber(object : OnResultCallBack<TestBean> {
                //getDatasNoCache
                override fun onSuccess(t: TestBean) {
                    t.list.forEach {
                        resultStr += it.toString()
                        result.text = resultStr
                    }
                }

                override fun onError(code: Int, errorMsg: String) {
                    result.text = "onError: code:$code  errorMsg:$errorMsg";
                }
            }, lifecycle), 1, 10, "json")
        }
    }
}

